test/elixir_analyzer/test_suite/rpn_calculator_output_test.exs

```diff
@@ -0,0 +1,230 @@
+
+          {:ok, equation}
+        end
+      end,
+      defmodule RPNCalculator.Output do
+        def write(resource, filename, equation) do
+          {:ok, file} = resource.open(filename)
+
+          with :ok <- IO.write(file, equation) do
+            result = {:ok, equation}
+          else
+            _ -> {:error, "Unable to write to resource"}
+          end
+
+          resource.close(file)
+          result
+        end
+      end,
+      defmodule RPNCalculator.Output do
+        def write(resource, filename, equation) do
+          {:ok, file} = resource.open(filename)
+
+          try do
+            IO.write(file, equation)
+          rescue
+            _ -> {:error, "Unable to write to resource"}
+          end
+
+          resource.close(file)
+          {:ok, equation}
+        end
+      end
+    ]
+  end
+
+  test_exercise_analysis "open in try",
+    comments_include: [Constants.rpn_calculator_output_open_before_try()] do
+    defmodule RPNCalculator.Output do
+      def write(resource, filename, equation) do
+        try do
+          {:ok, file} = resource.open(filename)
+          IO.write(file, equation)
+        rescue
+          _ -> {:error, "Unable to write to resource"}
+        else
+          :ok -> {:ok, equation}
+        after
+          resource.close(file)
+        end
+      end
+    end
+  end
+
+  test_exercise_analysis "write before try",
+    comments: [Constants.rpn_calculator_output_write_in_try()] do
+    [
+      defmodule RPNCalculator.Output do
+        def write(resource, filename, equation) do
+          {:ok, file} = resource.open(filename)
+          IO.write(file, equation)
+
+          try do
+            :ok
+          rescue
+            _ -> {:error, "Unable to write to resource"}
+          else
+            :ok -> {:ok, equation}
+          after
+            resource.close(file)
+          end
+        end
+      end,
+      defmodule RPNCalculator.Output do
+        def write(resource, filename, equation) do
+          {:ok, file} = resource.open(filename)
+          IO.write(file, equation)
+          :break
+
+          try do
+            :ok
+            :lol
+          rescue
+            _ -> {:error, "Unable to write to resource"}
+          else
+            :ok -> {:ok, equation}
+          after
+            resource.close(file)
+          end
+        end
+      end
+    ]
+  end
+
+  test_exercise_analysis "output not in else",
+    comments_include: [Constants.rpn_calculator_output_output_in_else()] do
+    defmodule RPNCalculator.Output do
+      def write(resource, filename, equation) do
+        {:ok, file} = resource.open(filename)
+
+        try do
+          IO.write(file, equation)
+        rescue
+          _ -> ok = false
+        else
+          :ok -> ok = true
+        after
+          resource.close(file)
+        end
+
+        if ok,
+          do: {:ok, equation},
+          else: {:error, "Unable to write to resource"}
+      end
+    end
```

Comment:
This code wouldn't compile 🤔 usually that doesn't matter in the test, but here I was looking at the failing example to figure out what exactly this check is supposed to be so that I can judge if the comment in website copy makes sense. I would propose this "wrong" (but passing) solution that doesn't even use `else`:

```suggestion
    defmodule RPNCalculator.Output do
      def write(resource, filename, equation) do
        {:ok, file} = resource.open(filename)

        result =
          try do
            IO.write(file, equation)
          rescue
            error -> error
          after
            resource.close(file)
          end

        case result do
          :ok -> {:ok, equation}
          _ -> {:error, "Unable to write to resource"}
        end
      end
    end
```